client/app/types/course/referenceTimelines.ts

```diff
@@ -0,0 +1,48 @@
+export interface TimeData {
+  id: number;
+  startAt: string;
+  bonusEndAt?: string;
+  endAt?: string;
+}
+
+export interface TimelineData {
+  id: number;
+  timesCount: number;
+  title: string;
+  default?: boolean;
+  weight?: number;
+  assignees?: number;
+}
+
+export interface ItemWithTimeData {
+  id: number;
+  title: string;
+  times: Record<TimelineData['id'], TimeData>;
```

Comment:
No. If you notice the response structure in `app/views/course/reference_timelines/index.json.jbuilder` and how this attribute was used in `ItemsSidebar` or `TimelinesStack`, `times` is correctly typed. This is because I need to get the time in a timeline for an item in O(1) time.

Reply:
You are right!